### PR TITLE
Fix issue#765: Logged out users can't mark favorite guides

### LINKED ISSUE #765: Non-LoggedIn User shouldn't be able to mark guide as FAVORITE
If a user is not logged-in and tries to access any guide url(`http://localhost:3000/en/guides/sky-blue-kemmer`) then he/she should not be able to mark the guide as `favorite`.

screen shot 2016-07-11 at 8 16 43 pm

### PR DESCRIPTION
**Pull request by TanSA05**
I added the following code:
* An `if` condition in `guides/show.js` to check if the user is logged in.
* Made some changes to `alertsService` to display an `alert` `msg_type` for appropriate requests.

This PR fixes #765

### CODE REVIEW

File: app/assets/javascripts/angular-libs/angular.openfarm.js [hunk 1]
```diff
@@ -27,6 +27,9 @@ openFarmModule.factory('alertsService', ['$rootScope',
         if (['200','201','202'].indexOf(code) !== -1) {
           msg_type = 'success'
         }
+        if (code % 400 > 0 && code % 400 < 100) {
```
Comment (simonv3) [inline app/assets/javascripts/angular-libs/angular.openfarm.js]: Why not include 400? Also why not just do `code >= 400 && code < 500`?

File: app/assets/javascripts/guides/show.js [hunk 2]
```diff
@@ -219,34 +221,38 @@ openFarmApp.controller('showGuideCtrl', ['$scope', '$http', 'guideService', '$q'
...
-
     function favoriteGuide (guideId) {
-      $scope.updatingFavoritedGuides = true;
-      var favorited_guide_ids = $scope.currentUser.favorited_guides.map(function (g) { return g.id; }) || [];
-      var index = favorited_guide_ids.indexOf(guideId);
-      if (index === -1) {
-        favorited_guide_ids.push(guideId);
-      } else {
-        favorited_guide_ids.splice(index, 1);
+      if ($scope.currentUser == null){
```
Comment (simonv3) [inline app/assets/javascripts/guides/show.js]: Yeah, please fix these.

Comment (TanSA05) [inline app/assets/javascripts/guides/show.js]: @simonv3 It stops working when I include three =

**Comment by simonv3** [comment]
👍  thanks for this Taneea!

**Comment by simonv3** [comment]
http://staging.openfarm.cc/en/guides/blah Works!

**Pull request merged**